### PR TITLE
Aryanmajeed/issue19

### PR DESCRIPTION
**Pull request by aryankarim**
creating this PR again because:

1. input of interests is now an array of strings
2. removed boilerplate code
2. added a custom hook for image upload validation

### CODE REVIEW

**Review by HaiderAltufaily — COMMENTED** [review]
Nice work @aryanmajeed, there is only one problem for the interests array that it submits the form whenever you press "Enter", Noora had the same issue and we could fix it with one line of code:

File: pages/setting/index.js [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    uploadInput.current.click();
+  }
+
+  const onChangeFile = (event) => {
+    setUploadedImage(event.target.files[0]);
+  };
+
+  const addItemToInterestArray = (e) => {
+    if (e.keyCode === 13 || e.type === "blur") {
+      setInterestsArray((prev) => [
```
Comment (HaiderAltufaily) [inline pages/setting/index.js:78]: Here before `setInterestArray` write `e.preventDefault()` and it should be fixed

Comment (aryankarim) [reply pages/setting/index.js:78]: Thanks! I will fix that!

File: pages/setting/index.js [hunk 2]
```diff
@@ -65,38 +66,60 @@ export default function Index() {
...
-  const handleInterestArray = (e) => {
-    if (e.keyCode === 13) {
-      setInterestsArray((prev) => [...prev, e.target.value]);
+  const deleteItemFromInterestArray = (id) => {
+    setInterestsArray((prev) => prev.filter((interest) => interest.id !== id));
+  };
+
+  //Below function will create all the input and textareas
+  //by iterating the inputs key located in public/locales/en/setting.json
+  //
```
Comment (MuhannedNoman) [inline pages/setting/index.js]: The comment doesn't add any value. Remove it.

**Review by MuhannedNoman — APPROVED** [review]
LGTM, nice work looping through the textfields.

File: pages/setting/index.js (hunk 2 above)
Comment (aryankarim) [reply pages/setting/index.js]: done!

**Review by HaiderAltufaily — APPROVED** [review]

**Pull request merged**